File: fixtures/php/controller-service-provider/src/PaymentController.php
<?php

namespace ChakraFixture\Api;

use ChakraFixture\Service\PaymentService;

final class PaymentController
{
    public function __construct(private PaymentService $service)
    {
    }

    public function refund(int $amountCents): void
    {
        $this->service->refund($amountCents);
    }
}

File: fixtures/java/controller-service-provider/src/main/java/chakra/payments/api/PaymentController.java
package chakra.payments.api;

import chakra.payments.service.PaymentService;

/** HTTP controller delegating refunds to the payment service. */
public class PaymentController {

    private final PaymentService service;

    public PaymentController(PaymentService service) {
        this.service = service;
    }

    public boolean refund(long amountCents) {
        return this.service.refund(amountCents);
    }
}
